components/collector/src/source_collectors/api_source_collectors/azure_devops/merge_requests.py

```diff
@@ -0,0 +1,66 @@
+"""Azure Devops Server merge requests collector."""
+
+from typing import cast
+
+from collector_utilities.functions import match_string_or_regular_expression
+from collector_utilities.type import URL
+from source_model import Entity, SourceMeasurement, SourceResponses
+
+from .base import AzureDevopsRepositoryBase
+
+
+class AzureDevopsMergeRequests(AzureDevopsRepositoryBase):
+    """Collector for merge requests (pull requests in Azure Devops)."""
+
+    async def _api_url(self) -> URL:
+        """Extend to add the pull requests API path."""
+        api_url = str(await super()._api_url())
+        return URL(f"{api_url}/pullrequests?api-version=4.1&searchCriteria.status=all")
+
+    async def _landing_url(self, responses: SourceResponses) -> URL:
+        """Extend to add the pull requests path."""
+        landing_url = str(await super()._landing_url(responses))
+        return URL(f"{landing_url}/pullrequests")
+
+    async def _parse_source_responses(self, responses: SourceResponses) -> SourceMeasurement:
+        """Override to parse the merge requests from the responses."""
+        merge_requests = []
+        for response in responses:
+            merge_requests.extend((await response.json())["value"])
+        landing_url = (await self._landing_url(responses)).rstrip("s")
+        entities = [
+            Entity(
+                key=merge_request["pullRequestId"],
+                title=merge_request["title"],
+                target_branch=merge_request["targetRefName"],
+                url=f"{landing_url}/{merge_request['pullRequestId']}",
+                state=merge_request["status"],
+                created=merge_request.get("creationDate"),
+                closed=merge_request.get("closedDate"),
+                downvotes=str(self._downvotes(merge_request)),
+                upvotes=str(self._upvotes(merge_request)),
+            )
+            for merge_request in merge_requests
+            if self._include_merge_request(merge_request)
+        ]
```

Comment:
For readability maybe use a simple for loop instead of list comprehension? Might be just personal preference though

Reply:
I've extracted the creation of the entity to a separate method so that the list comprehension fits on one line.